### ISSUE TITLE
Custom containers reset

### ISSUE THREAD

**Issue by attis** [mozilla/testpilot-containers#346]
Hi,

I just noticed all my custom containers were gone and were replaced with the default ones.
Might happened with the upgrade to FF 52.0.

**Comment by src-101** [comment]
Yup. Both my bug #349 and this are the same. You can mark anyone as a duplicate.

**Comment by groovecoder** [comment]
Thanks for the bug report. We'll triage at our Containers meeting tomorrow.

**Comment by klemens** [comment]
Mine were also lost after the update, but I could recover them by recreating them with the same name. All cookies etc. were still there.

**Comment by Myer** [comment]
Well, how do I find out how my custom containers were called? Can't remember it unfortunately...

**Comment by klemens** [comment]
It is displayed e.g. in the `Cookies` panel as the last property at the bottom when you select a specific cookie. Unfortunately it does not support sorting or searching and the developer tools don't yet display the Context/Container.

**Comment by Myer** [comment]
As mentioned above, if I for example check the "google.com" domain the last property of those cookies says "Pending to be deleted" which should be the name of my custom container for my two gmail-containers.

I just hope there is another place where the container names are stored...

**Comment by klemens** [comment]
The container names are stored in a `containers.json` file in the profile, where each container is assigned to the corresponding `userContextId`. These seem to be simply increasing, so you could create some containers and manually set their ids to some numbers 1 to x and then check the cookie panel again.

**Comment by groovecoder** [comment]
@bakulf can you check into this?

@attis, @src-101 - which OS are you using?

**Comment by attis** [comment]
macOS in my case. Let me know if you need me to be more specific :)

**Comment by src-101** [comment]
Windows 10 for me.

**Comment by Myer** [comment]
Also Windows 10 x64

**Comment by smichel17** [comment]
Fedora 25 (gnome 3.22.3, if that matters)

- Containers I had added were removed, and tabs that were in them during my previous session were moved into the default container
- Containers I had renamed were reset back to the default name and icon, but tabs in those containers remained

**Comment by SuperMatt** [comment]
I too have this bug on Fedora 25 and OSX (can't remember which version). I made a backup of my containers.json because I've been syncing them between machines. On the Mac, I found that after closing firefox, replacing containers.json and starting again, *sometimes* it would keep my settings. However, under Fedora 25 it always resets.

If I find anything, I will post it here.

**Comment by FreeWF** [comment]
Because of this bug, I lost more than 300 links, hours and hours of research during several months. I absolutely need to recover them. Does anybody knows in which file(s) all the links were saved ? I don't mind if the links are not anymore in the right containers, but I absolutely need to recover the links. Please help...

**Comment by AssafTzurEl** [comment]
Still happens, latest FF (updated to 66.0.5 a few days ago). Any news?

**Comment by lhorace** [comment]
@FreeWF You might find this useful #1409

**Comment by ALTanir** [comment]
@klemens> Mine were also lost ...recreating.. same ...still there.
Reply= [My story was same](https://github.com/mozilla/multi-account-containers/issues/1409#issuecomment-491914991)
Also the files- "containers.json" contained containers names and ID's and "cookies.sqlite"- that copy&backup to another mozilla(any) performs the similar logged in account properly !!
#1412
